lib/view/pages/events/view_events_page.dart

```diff
@@ -98,366 +96,163 @@ class _HomePageWidgetState extends State<ViewEvents> {
                       } else if (snapshot.hasError) {
                         return Text('Error: ${snapshot.error}');
                       } else if (snapshot.hasData) {
-                        allEvents = [];
-                        for (var x in snapshot.data!.docs) {
-                          var data = x.data() as Map<String, dynamic>;
-                          Event event = Event.fromMap(data, x.id);
-                          allEvents?.add(event);
+                        if (allEvents == null) {
```

Comment:
Unfortunately, this I believe makes the view events page less responsive. It does not update when you join an event or subscribe to an event. I tested it on my old code and it does update , but now it does not update anymore.

Reply:
Okay I'll take a look

Reply:
I just tested this on another branch it it gives the same issue...The problem is...the user does join the event, but it only shows when the page reloads

Reply:
On my side when I join an event,it immediatelly adds the event to the upcoming cardsat the end. Same with subscribe. I'm not too sure what branch you ran, but on my side it does immediatelly update when I join/subscribe in the sense tht there is a card added to the upcoming cards/subscribed cards. This is what I meant

Reply:
I agree that on the event it self the text/icon itself only updates when the page reloads, but the cards are immediatelly supposed to be added when a user joins/subscribes